pkg/cover/backend/mach-o.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package backend
+
+import (
+	"debug/macho"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/google/syzkaller/pkg/host"
+	"github.com/google/syzkaller/sys/targets"
+)
+
+func makeMachO(target *targets.Target, objDir, srcDir, buildDir string,
+	hostModules []host.KernelModule) (*Impl, error) {
+	// TODO: Linux/ELF supports module symbols. We should probably also do that
+	// for XNU/Mach-O. To maximize code re-use we already have a lot of the
+	// plumbing for module support. I think we mainly miss an equivalent to
+	// discoverModules and elfReadModuleCoverPoints at the moment.
+	if len(hostModules) != 0 {
```

Comment:
I think it makes sense to call discoverModules in makeDWARF. It already checks if OS is supported, produces and error and adds the main module.

Reply:
True, after having read discoverModules I agree. It even checks this as well :)